src/foam/dao/SQL.js

```diff
@@ -84,12 +84,17 @@ foam.CLASS({
 });
 
 
+/**
+   The base models Unary and Binary are used to validate the arguments
```

Comment:
Also, regardless of changing/removing the validation, this pull-request is a patch to fix the MLang SQL that is already part of the foam package.  So these fixes need to be applied and we should discuss the validation refactoring separately.